### PR TITLE
feat: add couchdb extractor

### PR DESCRIPTION
**Pull request by GrayFlash**

### CODE REVIEW

File: plugins/extractors/couchdb/couchdb.go [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+	Password string `mapstructure:"password" validate:"required"`
+	Host     string `mapstructure:"host" validate:"required"`
+}
+
+var sampleConfig = `
+ host: localhost:5984
+ user_id: admin
+ password: couchdb`
+
+// Extractor manages the extraction of data from MySQL
```
Comment (burntcarrot) [inline plugins/extractors/couchdb/couchdb.go:42]: Sorry for sounding picky, but I noticed that the comments still reflect `MySQL` as opposed to `CouchDB` while I was browsing the PR.

Similar occurrence at line 93.

Comment (GrayFlash) [reply plugins/extractors/couchdb/couchdb.go:42]: yup, missed changing that

File: plugins/extractors/couchdb/couchdb.go [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+		return
+	}
+
+	return
+}
+
+// Extract extracts the data from the CouchDB server
+// and collected through the out channel
+func (e *Extractor) Extract(ctx context.Context, emit plugins.Emit) (err error) {
+	defer e.client.Close(context.TODO())
```
Comment (StewartJingga) [inline plugins/extractors/couchdb/couchdb.go:96]: just use context from the argument instead of TODO. Applies to all occurrences.

**Review by StewartJingga — CHANGES_REQUESTED** [review]

File: plugins/extractors/couchdb/couchdb.go [hunk 3]
```diff
@@ -43,6 +43,7 @@ var sampleConfig = `
 type Extractor struct {
 	client      *kivik.Client
 	db          *kivik.DB
+	ctx         context.Context
```
Comment (StewartJingga) [inline plugins/extractors/couchdb/couchdb.go:46]: can we get by not storing context as internal state?

it is okay to keep passing context

Comment (GrayFlash) [reply plugins/extractors/couchdb/couchdb.go:46]: okay

**Review by StewartJingga — APPROVED** [review]

**Pull request merged**